lib/internal/url.js

```diff
@@ -1436,6 +1436,10 @@ function toPathIfFileURL(fileURLOrPath) {
   return fileURLToPath(fileURLOrPath);
 }
 
+function getURLOrigin(url) {
+  return bindingUrl.getOrigin(url);
```

Comment:
I think one would expect that method to support `URL` instances.
```suggestion
  return typeof url === 'string' ? bindingUrl.getOrigin(url) : url.origin;
```
```suggestion
  return bindingUrl.getOrigin(`${url}`);
```
But also, it's an internal function, so I guess it's OK to impose such restrictions given that we don't need it to support things that are not strings, but then we need to document that with JSDoc, and maybe change the function name to reflect that.

Reply:
I am not sure I understand the proposed change. If you are assuming that you are receiving a URL instance, then url.origin would be the correct function to call? bindingUrl.getOrigin looks like a function to avoid as much as possible.